src/Service/Resolvers/SqlQueryEngine.cs

```diff
@@ -213,6 +215,40 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
             // nextLink is the URL needed to get the next page of records using the same query options
             // with $after base64 encoded for opaqueness
             string path = UriHelper.GetEncodedUrl(_httpContextAccessor.HttpContext!.Request).Split('?')[0];
+
+            RuntimeConfig runtimeConfig = _runtimeConfigProvider.GetConfig();
+            // If the base route is not empty, we need to insert it into the URI before the rest path.
+            string? baseRoute = runtimeConfig.Runtime.Rest.BaseRoute;
+            if (!string.IsNullOrEmpty(baseRoute))
+            {
+                _sqlMetadataProvider.TryGetEntityPathFromName(context.EntityName, out string? pathNameForEntity);
+
+                // Path is of the form ....restPath/pathNameForEntity. We want to insert the base route before the restPath,
+                // hence we remove pathNameForEntity and restPath from the path and append them back after appending the base route to the path.
+                // Initialize lastIdxOfPathToBeIncluded by removing the pathNameForEntity from the path.
+                int lastIdxOfPathToBeIncluded = path.Length - pathNameForEntity!.Length - 1;
+
+                // Trim any forward slashes present at the end.
+                while (path[lastIdxOfPathToBeIncluded] == '/')
+                {
+                    lastIdxOfPathToBeIncluded--;
+                }
+
+                lastIdxOfPathToBeIncluded = lastIdxOfPathToBeIncluded - runtimeConfig.Runtime.Rest.Path.Length;
+
+                // Initialize pathWithBaseRoute by removing restPath and pathNameForEntity from the path.
+                // Finally, it will be of the form: .../baseRoute/restPath/pathNameForEntity.
+                StringBuilder pathWithBaseRoute = new(path.Substring(0, lastIdxOfPathToBeIncluded + 1));
+
+                // Append base route.
+                pathWithBaseRoute.Append(baseRoute);
+                // Append rest path.
+                pathWithBaseRoute.Append($"{runtimeConfig.Runtime.Rest.Path}");
```

Comment:
is runtimeConfig.Runtime.Rest.Path expected to contain a forward slash?

Reply:
Yes, that's correct. If it doesn't, we will fail during the config validation stage.